flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/util/OperationExecutorFactory.java

```diff
@@ -0,0 +1,136 @@
+
+/** Factory to create the operation executor. */
+public class OperationExecutorFactory {
+
+    public static Callable<ResultSet> createGetCatalogsExecutor(
+            SqlGatewayService service, SessionHandle sessionHandle) {
+        return () -> executeGetCatalogs(service, sessionHandle);
+    }
+
+    public static Callable<ResultSet> createGetSchemasExecutor(
+            SqlGatewayService service,
+            SessionHandle sessionHandle,
+            String catalogName,
+            String schemaName) {
+        return () -> executeGetSchemas(service, sessionHandle, catalogName, schemaName);
+    }
+
+    private static ResultSet executeGetCatalogs(
+            SqlGatewayService service, SessionHandle sessionHandle) {
+        Set<String> catalogNames = service.listCatalogs(sessionHandle);
+        return new ResultSet(
+                EOS,
+                null,
+                GET_CATALOGS_SCHEMA,
+                catalogNames.stream()
+                        .map(OperationExecutorFactory::packData)
+                        .collect(Collectors.toList()));
+    }
+
+    private static ResultSet executeGetSchemas(
+            SqlGatewayService service,
+            SessionHandle sessionHandle,
+            String catalogName,
+            String schemaName) {
+        Set<String> schemaNames =
+                filterStringSetBy(service.listDatabases(sessionHandle, catalogName), schemaName);
+
+        return new ResultSet(
+                EOS,
+                null,
+                GET_SCHEMAS_SCHEMA,
+                schemaNames.stream()
+                        .map(name -> packData(name, catalogName))
+                        .collect(Collectors.toList()));
+    }
+
+    // -------------------------------------
+    // useful methods
+    // -------------------------------------
+
+    /**
+     * Covert SQL 'like' pattern to a Java regular expression. Underscores (_) are converted to '.'
+     * and percent signs (%) are converted to '.*'. Note: escape characters are removed.
+     *
+     * @param pattern the SQL pattern to convert.
+     * @return the equivalent Java regular expression of the pattern.
+     */
+    private static String convertNamePattern(String pattern) {
+        if ((pattern == null) || pattern.isEmpty()) {
+            pattern = "%";
+        }
+        String wStr = ".*";
+        return pattern.replaceAll("([^\\\\])%", "$1" + wStr)
+                .replaceAll("\\\\%", "%")
+                .replaceAll("^%", wStr)
+                .replaceAll("([^\\\\])_", "$1.")
+                .replaceAll("\\\\_", "_")
+                .replaceAll("^_", ".");
+    }
+
+    public static Set<String> filterStringSetBy(Set<String> set, String pattern) {
+        return set.stream()
+                .filter(name -> name.matches(convertNamePattern(pattern)))
+                .collect(Collectors.toSet());
+    }
+
+    private static GenericRowData packData(List<Object> data) {
```

Comment:
It seems `packData(List<Object> data)` and `packData(Object... data)` are the same. We can combine into one.